azkaban-exec-server/src/test/java/azkaban/container/FlowContainerTest.java

```diff
@@ -99,23 +100,47 @@ public void setup() throws Exception {
     this.projectLoader = mock(ProjectLoader.class);
   }
 
-  /**
-   * FIXME: This test is incomplete for now as there is an expected merge conflict which will
-   * change interface with submitFlow. Once new code is merged in FlowContainer,
-   * this test will need to be completed by invoking a flow.
-   * @throws Exception
-   */
-  @Test
-  public void testExecSimple() throws Exception {
+  @After
+  public void destroy() {
+    if (this.flowContainer == null) {
+      return;
+    }
+    try {
+      this.flowContainer.closeMBeans();
+      this.flowContainer = null;
+    } catch (final Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void startFlowContainer() throws IOException {
     final ExecutableFlow execFlow = createTestExecutableFlowFromYaml("basicflowyamltest", "basic_flow");
     execFlow.setExecutionId(1);
     final ProjectFileHandler handler = new ProjectFileHandler(1, 1, 1, "testUser", "zip", "test.zip",
         1, null, null, null, "111.111.111.111");
     when(this.projectLoader.fetchProjectMetaData(anyInt(), anyInt())).thenReturn(handler);
 
-    final FlowContainer flowContainer = SERVICE_PROVIDER.getInstance(FlowContainer.class);
-    flowContainer.start(props);
-    FlowContainer.launchCtrlMsgListener(flowContainer);
+    this.flowContainer = SERVICE_PROVIDER.getInstance(FlowContainer.class);
+    this.flowContainer.start(props);
+  }
+
+  /**
+   * Don't like the name? Me neither - DJ
+   * Test if Callback Manager is not created if not enabled.
+   * This test must be the first as Callback Manager is a static object which once created
+   * remains there for the remainder of the test suite. The name and ordering ensures that
+   * this test always runs first.
+   */
+  @Test(expected = IllegalStateException.class)
+  public void aaa() throws Exception {
```

Comment:
Why don't you do it in startup? There is already a method in this class for the same called setUp. Why don't you use that?
For e.g. https://junit.org/junit4/javadoc/4.12/org/junit/BeforeClass.html

Reply:
We need to do it once for this negative test only. How will putting it in setUp help? It is called everyday.